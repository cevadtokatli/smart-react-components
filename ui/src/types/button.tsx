import { IntrinsicStyledCoreProps } from '@smart-react-components/core/element-props/intrinsic-styled-core-props'
import { ContentElement, JSXElementProps, PaletteProp, Partial, ResponsiveProp, ShapeProp, SizeProp } from '@smart-react-components/core/types'

export interface ButtonGenericProps<T extends ContentElement> extends
  IntrinsicStyledCoreProps,
  Partial<ResponsiveProp<'size', SizeProp>> {
  children: T | T[]
  elementProps?: JSXElementProps
  hasHover?: boolean
  hasSpace?: boolean
  hasWaveEffect?: boolean
  isActive?: boolean
  isBlock?: boolean
  isDisabled?: boolean
  isFixedSize?: boolean
  isLink?: boolean
  isLoading?: boolean
  isSoft?: boolean
  isOutline?: boolean
  leftAddon?: JSX.Element
  loading?: JSX.Element
  palette?: PaletteProp
  rightAddon?: JSX.Element
  shape?: ShapeProp
  waveEffectPalette?: PaletteProp
}

export interface ButtonListGenericProps extends
  IntrinsicStyledCoreProps,
  Partial<ResponsiveProp<'size', SizeProp>> {
  children: JSX.Element[]
  elementProps?: JSXElementProps
  hasHover?: boolean
  hasWaveEffect?: boolean
  isBlock?: boolean
  isFixedSize?: boolean
  isLink?: boolean
  isSoft?: boolean
  isOutline?: boolean
  palette?: PaletteProp
  shape?: ShapeProp
  waveEffectPalette?: PaletteProp
}

export enum ButtonType {
  BUTTON = 'button',
  RESET = 'reset',
  SUBMIT = 'submit',
}
