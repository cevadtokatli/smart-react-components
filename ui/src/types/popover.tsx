import { Position } from './'

export interface PopoverCalculationResult {
  arrowStyle: string
  isArrowInHeader: boolean
  overflow: number
  pos: Position
  style: string
}
