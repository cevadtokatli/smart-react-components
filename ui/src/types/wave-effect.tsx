export interface EffectItem {
  element: JSX.Element
  endTime: number
}
