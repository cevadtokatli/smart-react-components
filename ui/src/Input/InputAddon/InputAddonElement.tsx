import Div from '@smart-react-components/core/Element/Div'
import Label from '@smart-react-components/core/Element/Label'
import { StyledProps } from '@smart-react-components/core/styled-props'
import { PaletteProp, ResponsiveProp, ShapeProp, SizeProp } from '@smart-react-components/core/types'
import styled from 'styled-components'
import { OrderPosition } from '../../types'
import { toCSSValue } from '../../util/css'
import { getReverseOrderPosition } from '../../util/props'

interface Props extends
  StyledProps,
  ResponsiveProp<'inputSize', SizeProp> {
  addonPosition: OrderPosition
  hasBorder: boolean
  isDisabled: boolean
  isFocused: boolean
  isOutline: boolean
  isSeparated: boolean
  isSoft: boolean
  palette: PaletteProp
  shape?: ShapeProp
}

export default styled(Div).attrs<Props>(({ isSeparated }: Props) => ({
  ...(isSeparated && { as: Label }),
  getInputSize: (v, t) => `
    font-size: ${t.$.size.form.blockLabel[v].fontSize};
    padding: ${isSeparated ? t.$.size.input[v].padding.y : 0} ${toCSSValue(t.$.size.input[v].padding.x)(v => v / 2)};
  `,
}))<Props>(({ theme, addonPosition, hasBorder, isDisabled, isFocused, isOutline, isSeparated, isSoft, palette, shape }: Props) => `
  align-items: center;
  border-bottom-${OrderPosition[addonPosition].toLowerCase()}-radius: ${theme.$.radius.input[shape]};
  border-top-${OrderPosition[addonPosition].toLowerCase()}-radius: ${theme.$.radius.input[shape]};
  box-sizing: border-box;
  display: inline-flex;
  flex: 0 0 auto;
  justify-content: center;
  transition: 200ms 0s ease-in-out;
  transition-property: background, border, color, fill;

  ${isSeparated
    ? `
      align-self: flex-start;
    `
    : ''
  }

  ${hasBorder
    ? `
      border: solid 1px;
      border-${OrderPosition[getReverseOrderPosition(addonPosition)].toLowerCase()}: 0;
    `
    : ''
  }

  ${!isOutline
    ? `
      ${!isSoft
        ? `
          color: ${theme.$.palette[palette].font};
          fill: ${theme.$.palette[palette].font};

          ${(isSeparated || !isFocused)
            ? `
              background: ${theme.$.palette[palette].main};
              border-color: ${theme.$.palette[palette].dark};
            `
            : `
              background: ${theme.$.palette[palette].dynamicer};
              border-color: ${theme.$.palette[palette].dynamicest};
            `
          }
        `
        : `
          color: ${theme.$.palette[palette].softFont};
          fill: ${theme.$.palette[palette].softFont};

          ${(isSeparated || !isFocused)
            ? `
              background: ${theme.$.palette[palette].soft};
              border-color: ${theme.$.palette[palette].softDark};
            `
            : `
              background: ${theme.$.palette[palette].softDarker};
              border-color: ${theme.$.palette[palette].softDarkest};
            `
          }
        `
      }
    `
    : `
      background: ${theme.$.color.dynamic.background};
      border-color ${(isSeparated || !isFocused) ? theme.$.color.dynamic.accent : theme.$.palette[palette].main};
      color: ${theme.$.color.dynamic.font};
      fill: ${theme.$.color.dynamic.font};
    `
  }

  ${(!isSeparated && isDisabled)
    ? `
      opacity: ${theme.$.opacity.formDisabled};

      ${isOutline
        ? `
          background: ${theme.$.color.dynamic.accent};  
          color: ${theme.$.color.dynamic.gray};
        `
        : ''
      }
    `
    : ''
  }
`)
