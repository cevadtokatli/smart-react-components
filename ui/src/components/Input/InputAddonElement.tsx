import Div from '@smart-react-components/core/Element/Div'
import Svg from '@smart-react-components/core/Element/Svg'
import { StyledProps } from '@smart-react-components/core/styled-props'
import { PaletteProp, ResponsiveProp, ShapeProp, SizeProp } from '@smart-react-components/core/types'
import styled from 'styled-components'
import { OrderPosition } from '../../types'
import { toCSSValue } from '../../util/css'
import { getReverseOrderPosition } from '../../util/props'

interface Props extends
  StyledProps,
  ResponsiveProp<'inputSize', SizeProp> {
  addonPosition: OrderPosition
  cursorKey: string
  hasBorder: boolean
  isDisabled: boolean
  isExcluded: boolean
  isFocused: boolean
  isOutline: boolean
  isSeparated: boolean
  isSoft: boolean
  palette: PaletteProp
  shape?: ShapeProp
}

export default styled(Div).attrs<Props>(({ className = '', isSeparated }) => ({
  ...(isSeparated && { as: 'label' }),
  className: `src-input-addon ${className}`,
  getInputSize: (v, t) => `
    font-size: ${t.$.size.form.blockLabel[v].fontSize};
    padding: ${toCSSValue(t.$.size.input[v].padding.x)(v => v / 2)};

    ${Svg} {
      height: ${t.$.size.icon[v]};
      width: ${t.$.size.icon[v]};
    }
  `,
}))<Props>(({ theme, addonPosition, cursorKey, hasBorder, isDisabled, isExcluded, isFocused, isOutline, isSeparated, isSoft, palette, shape }: Props) => `
  align-items: center;
  box-sizing: border-box;
  cursor: ${theme.$.cursor[cursorKey]};
  display: inline-flex;
  fill: currentcolor;
  flex: 0 0 auto;
  justify-content: center;
  transition: 200ms 0s ease-in-out;
  transition-property: background, border, color, fill;

  ${isExcluded
    ? `
      border-bottom-${OrderPosition[addonPosition].toLowerCase()}-radius: ${theme.$.radius.input[shape]};
      border-top-${OrderPosition[addonPosition].toLowerCase()}-radius: ${theme.$.radius.input[shape]};

      ${hasBorder
        ? `
          border: solid 1px;
          border-${OrderPosition[getReverseOrderPosition(addonPosition)].toLowerCase()}: 0;
        `
        : ''
      }
    `
    : ''
  }

  ${!isOutline
    ? `
      ${!isSoft
        ? `
          color: ${theme.$.palette[palette].font};

          ${!isFocused
            ? `
              background: ${theme.$.palette[palette].main};
              border-color: ${theme.$.palette[palette].dark};
            `
            : `
              background: ${theme.$.palette[palette].dynamicer};
              border-color: ${theme.$.palette[palette].dynamicest};
            `
          }
        `
        : `
          color: ${theme.$.palette[palette].softFont};

          ${!isFocused
            ? `
              background: ${theme.$.palette[palette].soft};
              border-color: ${theme.$.palette[palette].softDark};
            `
            : `
              background: ${theme.$.palette[palette].softDarker};
              border-color: ${theme.$.palette[palette].softDarkest};
            `
          }
        `
      }
    `
    : `
      background: ${theme.$.color.dynamic.background};
      border-color ${!isFocused ? theme.$.color.dynamic.accent : theme.$.palette[palette].main};
      color: ${theme.$.color.dynamic.font};
    `
  }

  ${(!isSeparated && isDisabled)
    ? `
      opacity: ${theme.$.opacity.formDisabled};

      ${isOutline
        ? `
          background: ${theme.$.color.dynamic.accent};  
          color: ${theme.$.color.dynamic.gray};
        `
        : ''
      }
    `
    : ''
  }

  ${isDisabled
    ? `
      pointer-events: none;
    `
    : ''
  }
`)
