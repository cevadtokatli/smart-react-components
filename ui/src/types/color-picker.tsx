import { Color } from '@smart-react-components/core/types'
import { MutableRefObject } from 'react'

export enum ColorPickerFormat {
  HEX,
  RGB,
  HSL,
}

export interface PickerProps {
  color: MutableRefObject<Color>
  onDrag: () => void
  onEnd: (onMove: (e: Event) => void, onEnd: () => void) => void
  onRef?: (v: PickerRef) => void
  onStart: (onMove: (e: Event) => void, onEnd: () => void) => void
}

export interface PickerRef {
  onChange: () => void
  onExternalDrag?: () => void
}
