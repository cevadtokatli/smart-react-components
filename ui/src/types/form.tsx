import { IntrinsicStyledProps } from '@smart-react-components/core/element-props/intrinsic-styled-props'
import { ContentElement, JSXElementProps, PaletteProp, Partial, ResponsiveProp, SetState, ShapeProp, SizeProp } from '@smart-react-components/core/types'
import { FormValue } from '../types'

export interface GenericProps extends
  Partial<ResponsiveProp<'size', SizeProp>>,
  IntrinsicStyledProps {
  active: FormValue | FormValue[]
  children: JSX.Element | JSX.Element[]
  elementProps?: JSXElementProps
  hasBorder?: boolean
  hasHover?: boolean
  hasWaveEffect?: boolean
  isBlock?: boolean
  isDisabled?: boolean
  isRequired?: boolean
  isOutline?: boolean
  isSoft?: boolean
  label?: ContentElement
  leftAddon?: ContentElement
  leftIcon?: ContentElement
  palette?: PaletteProp
  rightAddon?: ContentElement
  rightIcon?: ContentElement
  shape?: ShapeProp
  setActive: SetState<FormValue | FormValue[]>
  waveEffectPalette?: PaletteProp
}
