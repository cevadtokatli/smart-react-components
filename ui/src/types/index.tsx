import { Value } from '@smart-react-components/core/types'

export type FormValue = Value | boolean

export enum InputType {
  EMAIL = 'email',
  PASSWORD = 'password',
  SEARCH = 'search',
  TEL = 'tel',
  TEXT = 'text',
  URL = 'url',
}

export enum OrderPosition {
  LEFT = 1,
  RIGHT = 2,
}

export enum Position {
  LEFT = 1,
  RIGHT = 2,
  TOP = 4,
  BOTTOM = 8,
}

export enum TriggerInteraction {
  NONE = 0,
  CLICK = 1,
  HOVER = 2,
  RIGHT_CLICK = 4,
}
