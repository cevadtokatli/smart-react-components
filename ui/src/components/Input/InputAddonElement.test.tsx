import { wrapTheme } from '@smart-react-components/core/test'
import { render } from '@testing-library/react'
import React from 'react'
import { OrderPosition } from '../../types'
import InputAddonElement from './InputAddonElement'

describe('<InputAddonElement />', () => {
  let props

  beforeAll(() => {
    props = {
      addonPosition: OrderPosition.LEFT,
      hasBorder: true,
      inputSize: 'medium',
      isDisabled: false,
      isFocused: false,
      isOutline: true,
      isSeparated: false,
      isSoft: false,
      palette: 'primary',
      shape: 'rectangle',
    }
  })

  it('should render component', () => {
    const { asFragment } = render(wrapTheme(<InputAddonElement {...props} />))
    expect(asFragment()).toMatchSnapshot()
  })

    it('should render component focused', () => {
    const { asFragment } = render(wrapTheme(<InputAddonElement {...props} isFocused />))
    expect(asFragment()).toMatchSnapshot()
  })

    it('should render component disabled', () => {
    const { asFragment } = render(wrapTheme(<InputAddonElement {...props} isDisabled />))
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render component not outline', () => {
    const { asFragment } = render(wrapTheme(<InputAddonElement {...props} isOutline={false} />))
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render component with soft colors', () => {
    const { asFragment } = render(wrapTheme(<InputAddonElement {...props} isOutline={false} isSoft />))
    expect(asFragment()).toMatchSnapshot()
  })

  it('should resize component responsively', () => {
    const { asFragment } = render(wrapTheme(<InputAddonElement {...props} inputSize="small" inputSizeMd="medium" inputSizeLg="large" />))
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render component at right', () => {
    const { asFragment } = render(wrapTheme(<InputAddonElement {...props} addonPosition={OrderPosition.RIGHT} />))
    expect(asFragment()).toMatchSnapshot()
  })

  it('should be focused', () => {
    const { asFragment } = render(wrapTheme(<InputAddonElement {...props} isFocused />))
    expect(asFragment()).toMatchSnapshot()
  })

  it('should be disabled', () => {
    const { asFragment } = render(wrapTheme(<InputAddonElement {...props} isDisabled />))
    expect(asFragment()).toMatchSnapshot()
  })

  it('should be separated', () => {
    const { asFragment } = render(wrapTheme(<InputAddonElement {...props} isSeparated />))
    expect(asFragment()).toMatchSnapshot()
  })
})
