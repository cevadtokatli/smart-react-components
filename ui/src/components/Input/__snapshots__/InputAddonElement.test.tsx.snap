// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputAddonElement /> should be disabled 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  padding: 0.45rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  fill: currentcolor;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: 200ms 0s ease-in-out;
  transition: 200ms 0s ease-in-out;
  -webkit-transition-property: background,border,color,fill;
  transition-property: background,border,color,fill;
  background: #fff;
  border-color: #dbdbdb;
  color: #323a46;
  opacity: 0.7;
  pointer-events: none;
  background: #dbdbdb;
  color: #eee;
}

<div
    class="c0 c1 src-input-addon "
    shape="rectangle"
  />
</DocumentFragment>
`;

exports[`<InputAddonElement /> should be focused 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  padding: 0.45rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  fill: currentcolor;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: 200ms 0s ease-in-out;
  transition: 200ms 0s ease-in-out;
  -webkit-transition-property: background,border,color,fill;
  transition-property: background,border,color,fill;
  background: #fff;
  border-color: #0665d0;
  color: #323a46;
}

<div
    class="c0 c1 src-input-addon "
    shape="rectangle"
  />
</DocumentFragment>
`;

exports[`<InputAddonElement /> should be separated 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  padding: 0.45rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  fill: currentcolor;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: 200ms 0s ease-in-out;
  transition: 200ms 0s ease-in-out;
  -webkit-transition-property: background,border,color,fill;
  transition-property: background,border,color,fill;
  border-bottom-left-radius: .2rem;
  border-top-left-radius: .2rem;
  border: solid 1px;
  border-right: 0;
  background: #fff;
  border-color: #dbdbdb;
  color: #323a46;
}

<label
    class="c0 c1 src-input-addon "
    shape="rectangle"
  />
</DocumentFragment>
`;

exports[`<InputAddonElement /> should render component 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  padding: 0.45rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  fill: currentcolor;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: 200ms 0s ease-in-out;
  transition: 200ms 0s ease-in-out;
  -webkit-transition-property: background,border,color,fill;
  transition-property: background,border,color,fill;
  background: #fff;
  border-color: #dbdbdb;
  color: #323a46;
}

<div
    class="c0 c1 src-input-addon "
    shape="rectangle"
  />
</DocumentFragment>
`;

exports[`<InputAddonElement /> should render component at right 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  padding: 0.45rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  fill: currentcolor;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: 200ms 0s ease-in-out;
  transition: 200ms 0s ease-in-out;
  -webkit-transition-property: background,border,color,fill;
  transition-property: background,border,color,fill;
  background: #fff;
  border-color: #dbdbdb;
  color: #323a46;
}

<div
    class="c0 c1 src-input-addon "
    shape="rectangle"
  />
</DocumentFragment>
`;

exports[`<InputAddonElement /> should render component disabled 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  padding: 0.45rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  fill: currentcolor;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: 200ms 0s ease-in-out;
  transition: 200ms 0s ease-in-out;
  -webkit-transition-property: background,border,color,fill;
  transition-property: background,border,color,fill;
  background: #fff;
  border-color: #dbdbdb;
  color: #323a46;
  opacity: 0.7;
  pointer-events: none;
  background: #dbdbdb;
  color: #eee;
}

<div
    class="c0 c1 src-input-addon "
    shape="rectangle"
  />
</DocumentFragment>
`;

exports[`<InputAddonElement /> should render component focused 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  padding: 0.45rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  fill: currentcolor;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: 200ms 0s ease-in-out;
  transition: 200ms 0s ease-in-out;
  -webkit-transition-property: background,border,color,fill;
  transition-property: background,border,color,fill;
  background: #fff;
  border-color: #0665d0;
  color: #323a46;
}

<div
    class="c0 c1 src-input-addon "
    shape="rectangle"
  />
</DocumentFragment>
`;

exports[`<InputAddonElement /> should render component not outline 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  padding: 0.45rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  fill: currentcolor;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: 200ms 0s ease-in-out;
  transition: 200ms 0s ease-in-out;
  -webkit-transition-property: background,border,color,fill;
  transition-property: background,border,color,fill;
  background: #0665d0;
  border-color: #055BBB;
  color: #fff;
}

<div
    class="c0 c1 src-input-addon "
    shape="rectangle"
  />
</DocumentFragment>
`;

exports[`<InputAddonElement /> should render component with soft colors 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  padding: 0.45rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  fill: currentcolor;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: 200ms 0s ease-in-out;
  transition: 200ms 0s ease-in-out;
  -webkit-transition-property: background,border,color,fill;
  transition-property: background,border,color,fill;
  background: #cde0f5;
  border-color: #b9d3f1;
  color: #033c7c;
}

<div
    class="c0 c1 src-input-addon "
    shape="rectangle"
  />
</DocumentFragment>
`;

exports[`<InputAddonElement /> should resize component responsively 1`] = `
<DocumentFragment>
  .c0 {
  font-size: .875rem;
  padding: 0.4rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  fill: currentcolor;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: 200ms 0s ease-in-out;
  transition: 200ms 0s ease-in-out;
  -webkit-transition-property: background,border,color,fill;
  transition-property: background,border,color,fill;
  background: #fff;
  border-color: #dbdbdb;
  color: #323a46;
}

@media (min-width:768px) {
  .c0 {
    font-size: 1rem;
    padding: 0.45rem;
  }
}

@media (min-width:992px) {
  .c0 {
    font-size: 1rem;
    padding: 0.5rem;
  }
}

<div
    class="c0 c1 src-input-addon "
    shape="rectangle"
  />
</DocumentFragment>
`;
